cmd/sops/subcommand/exec/exec.go

```diff
@@ -55,7 +73,11 @@ func ExecWithFile(opts ExecOpts) error {
 	if opts.Fifo {
 		// fifo handling needs to be async, even opening to write
 		// will block if there is no reader present
-		filename = GetPipe(dir, opts.Filename)
+		filename = opts.Filename
+		if filename == "" {
+			filename = FallbackFilename
+		}
+		filename = GetPipe(dir, filename)
 		go WritePipe(filename, opts.Plaintext)
 	} else {
 		handle := GetFile(dir, opts.Filename)
```

Comment:
Might be worth leaving a comment as to why this is `opts.Filename` vs `filename` to avoid future confusion (I could see myself changing this in the future 😅 )

Alternatively, you could pass `filename` to `GetFile` and check for `filename == FallbackFilename`, but then there is the edge-case where a user intentionally passes `"tmp-file"` as the filename

Reply:
Good point, I've updated the PR accordingly (and rebased while at it).

> Alternatively, you could pass `filename` to `GetFile` and check for `filename == FallbackFilename`, but then there is the edge-case where a user intentionally passes `"tmp-file"` as the filename

Yeah, that's why I did it this way ;-)